src/modules/0.0.21/aws_codebuild/mappers/build.ts

```diff
@@ -0,0 +1,205 @@
+    out.awsId = s.id;
+    out.buildStatus = s.buildStatus as BuildStatus;
+    if (!Object.values(ctx.memo?.cloud?.CodebuildProject ?? {}).length) {
+      out.project =
+        (await this.module.project.db.read(ctx, s.projectName)) ??
+        (await this.module.project.cloud.read(ctx, s.projectName));
+    } else {
+      out.project =
+        (await this.module.project.db.read(ctx, s.projectName)) ??
+        ctx?.memo?.cloud?.CodebuildProject?.[s.projectName ?? ''];
+    }
+    out.buildNumber = s.buildNumber;
+    out.endTime = s.endTime;
+    out.startTime = s.startTime;
+    return out;
+  }
+
+  async waitForBuildsToStop(client: CodeBuild, ids: string[]) {
+    // wait for policies to be attached
+    const input: BatchGetBuildsCommandInput = {
+      ids,
+    };
+    await createWaiter<CodeBuild, BatchGetBuildsCommandInput>(
+      {
+        client,
+        // all in seconds
+        maxWaitTime: 900,
+        minDelay: 1,
+        maxDelay: 4,
+      },
+      input,
+      async (cl, cmd) => {
+        try {
+          const data = await cl.batchGetBuilds(cmd);
+          const done = data?.builds?.every(bd => bd.buildStatus === BuildStatus.STOPPED || !!bd.endTime);
+          if (done) {
+            return { state: WaiterState.SUCCESS };
+          }
+          return { state: WaiterState.RETRY };
+        } catch (e: any) {
+          throw e;
+        }
+      },
+    );
+  }
+
+  listBuildIds = paginateBuilder<CodeBuild>(paginateListBuilds, 'ids');
+
+  getBuilds = crudBuilderFormat<CodeBuild, 'batchGetBuilds', Build[] | undefined>(
+    'batchGetBuilds',
+    input => input,
+    res => res?.builds,
+  );
+
+  stopBuild = crudBuilder2<CodeBuild, 'stopBuild'>('stopBuild', input => input);
+
+  deleteBuilds = crudBuilder2<CodeBuild, 'batchDeleteBuilds'>('batchDeleteBuilds', input => input);
+
+  cloud: Crud2<CodebuildBuildList> = new Crud2({
+    create: async (es: CodebuildBuildList[], ctx: Context) => {
+      // Do not cloud create, just restore database
+      await this.module.buildList.db.delete(es, ctx);
+    },
+    read: async (ctx: Context, id?: string) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+      let input: BatchGetBuildsCommandInput;
+      let builds: Build[] | undefined;
+      if (id) {
+        input = {
+          ids: [id],
+        };
+        builds = await this.getBuilds(client.cbClient, input);
+        if (!builds || builds.length !== 1) return;
+        return this.buildListMapper(builds[0], ctx);
+      }
+      const ids = await this.listBuildIds(client.cbClient);
+      if (!ids || !ids.length) return;
+      input = {
+        ids,
+      };
+      builds = await this.getBuilds(client.cbClient, input);
+      if (!builds) return;
+      const out = [];
+      for (const build of builds) {
+        const outB = await this.buildListMapper(build, ctx);
+        if (outB) out.push(outB);
+      }
+      return out;
+    },
+    updateOrReplace: () => 'update',
```

Comment:
Nit. This isn't necessary since this is the exact function auto-provided for you if you don't define the `updateOrReplace` method.

Reply:
got it, copied it from another module